base/libgit2/types.jl

```diff
@@ -670,32 +670,46 @@ function securezero!(cred::UserPasswordCredentials)
     return cred
 end
 
+function Base.:(==)(a::UserPasswordCredentials, b::UserPasswordCredentials)
+    a.user == b.user && a.pass == b.pass
+end
+
 "SSH credentials type"
 mutable struct SSHCredentials <: AbstractCredentials
     user::String
     pass::String
-    pubkey::String
     prvkey::String
+    pubkey::String
```

Comment:
Not strictly necessary for this PR but makes the field order match the constructor order. Since this type has inner constructors is it even possible to construct this type directly?